use-subscription/index.js

```diff
@@ -48,14 +49,25 @@ function useSubscription (channels, opts = {}) {
     .join(' ')
 
   useEffect(() => {
+    let timeout
     let ignoreResponce = false
-    changeSubscribing(true)
+
+    if (debounce > 0) {
+      timeout = setTimeout(() => {
+        changeSubscribing(true)
+      }, debounce)
+    } else {
+      changeSubscribing(true)
+    }
+
     add(store, subscriptions).then(() => {
+      timeout && clearTimeout(timeout)
       if (!ignoreResponce) changeSubscribing(false)
     })
     return () => {
       ignoreResponce = true
       remove(store, subscriptions)
+      timeout && clearTimeout(timeout)
```

Comment:
```suggestion
      if (timeout) clearTimeout(timeout)
```

Reply:
Should we move `clearTimeout` to some function which will also set `timeout = null`?

Reply:
Oops 😬